### PR TITLE
[Matrix] remove use of p8-platform and kodi-platform and allow Ubuntu build in travis

### PR DESCRIPTION
**Pull request by AlwinEsch**
About kodi-platform was on this addon not used and with p8-platform are now helpers on kodi-dev-kit folder where brings required thread support.

Also is with add of "#undef CreateDirectory" a fix about Windows build added. 

Further should with the new headers also the build for C++17 fixed.

And second commit to allow Ubuntu package test build within Travis CI

**WARNING** With missing backend no complete runtime test performed.

### CODE REVIEW

File: src/Freebox.cpp [hunk 1]
```diff
@@ -297,15 +302,14 @@ bool Freebox::StartSession ()
     #define HOST_NAME_MAX 256
   #endif
 #endif
-      char hostname [HOST_NAME_MAX + 1];
```
Comment (aassif) [inline src/Freebox.cpp:300]: Thanks to `GetHostname`, `HOST_NAME_MAX` can be removed entirely :)

**Comment by aassif** [comment]
Thanks! Very nice!

File: src/Freebox.cpp (hunk 1 above)
Comment (aassif) [reply src/Freebox.cpp:300]: Are you willing to do so before I merge?

Comment (AlwinEsch) [reply src/Freebox.cpp:300]: Ah yes, them no more need then :+1: becomes removed.

Comment (AlwinEsch) [reply src/Freebox.cpp:300]: Is done.

**Pull request merged**